### PR TITLE
Update CSharp highlights.

### PR DESCRIPTION
**Pull request by cranberry-knight**
Syntax highlighting update for C# language.

Changelist:
1. Added C#9 keywords:
   - `record`
   - `init`
   - `and`, `or`, `not` used in a pattern matching.
2. Added missing keywords from previous language versions:
    - `implicit`, `explicit`;
    - `internal`;
    - `event`, `add`, `remove`;
    - `checked`, `unchecked`, `fixed`.
3. Updated inline LINQ expressions with `let` and `equal`.
4. Highlighted `global` as a keyword when access to the global namespace.
5. Highlighted `value` as a keyword in filed accessors (`get`, `set`).

Used reference https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/

### CODE REVIEW

File: queries/c_sharp/highlights.scm [hunk 1]
```diff
@@ -1,5 +1,9 @@
 (identifier) @variable
 
+((identifier) @keyword @_p
+  (#eq? @_p "value")
```
Comment (stsewd) [inline queries/c_sharp/highlights.scm:4]: You can use `@keyword` instead of creating a new caputre

Comment (cranberry-knight) [reply queries/c_sharp/highlights.scm:4]: `value` is the contextual keyword (see [more](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/value))
For example:
```CSharp
class Foo
{
    private int number;
    public int Property
    {
        get => number;
        set {number = value;} // "value" is keyword
    }

    public void Bar()
    {
        var value = 10;
        var another = value; // "value" is simple variable name
    }
}
```

Therefore it should be highlighted when placed in the `accessor_declaration` subtree.

Or the comment is about simplifying the query and am I missing something?

Comment (cranberry-knight) [reply queries/c_sharp/highlights.scm:4]: Just found out that `get`, `set`, `add`, `remove` and `init` are also valid names for variables. As well as `and` and `or` and `record`

Comment (stsewd) [reply queries/c_sharp/highlights.scm:4]: what I mean is that you can write this query as

```
((identifier) @keyword
  (#eq? @keyword "value")
```

there is no need to create `@_p`

Comment (cranberry-knight) [reply queries/c_sharp/highlights.scm]: Got it! Removed redundant `@_p` capture.

Also checked contextual keywords highlighting (wrote about them earlier) and they are all good. Variables like `set` are highlighted properly.

**Review by vigoux — APPROVED** [review]

**Review by stsewd — APPROVED** [review]

**Pull request merged**